File: backend/NordicWebHub.Api/Models/Project.cs
using NordicWebHub.Api.Models.Enums;

namespace NordicWebHub.Api.Models;

public class Project
{
    public int Id { get; set; }

    public int CompanyId { get; set; }

    public Company Company { get; set; } = null!;

    public int? ProjectRequestId { get; set; }

    public ProjectRequest? ProjectRequest { get; set; }

    public string Title { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public ProjectStatus Status { get; set; } = ProjectStatus.Planned;

    public DateTime StartDate { get; set; }

    public DateTime Deadline { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/nordicwebhub-client/src/types/project.ts
export type ProjectStatus =
  | 'Planning'
  | 'Design'
  | 'Development'
  | 'Review'
  | 'Live'
  | 'Completed'

export type Project = {
  id: number
  companyId: number
  companyName: string
  projectRequestId: number | null
  projectRequestTitle: string
  title: string
  description: string
  status: ProjectStatus
  startDate: string
  deadline: string
  createdAt: string
}

export type CreateProjectDto = {
  companyId?: number
  projectRequestId?: number | null
  title: string
  description: string
  status: ProjectStatus
  startDate: string
  deadline: string
}

export type UpdateProjectDto = {
  companyId: number
  projectRequestId?: number | null
  title: string
  description: string
  startDate: string
  deadline: string
}

export type UpdateProjectStatusDto = {
  status: ProjectStatus
}
